pkg/util/push/push.go

```diff
@@ -6,19 +6,28 @@ import (
 
 	"github.com/go-kit/kit/log/level"
 	"github.com/weaveworks/common/httpgrpc"
+	"github.com/weaveworks/common/middleware"
+	"github.com/weaveworks/common/server"
 
 	"github.com/cortexproject/cortex/pkg/distributor"
 	"github.com/cortexproject/cortex/pkg/ingester/client"
 	"github.com/cortexproject/cortex/pkg/util"
 )
 
 // Handler is a http.Handler which accepts WriteRequests.
-func Handler(cfg distributor.Config, push func(context.Context, *client.WriteRequest) (*client.WriteResponse, error)) http.Handler {
+func Handler(cfg distributor.Config, serverCfg server.Config, push func(context.Context, *client.WriteRequest) (*client.WriteResponse, error)) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		logger := util.WithContext(ctx, util.Logger)
+		sourceIPs, err := middleware.NewSourceIPs(serverCfg.LogSourceIPsHeader, serverCfg.LogSourceIPsRegex)
```

Comment:
**thought**: Can we create this before we return the function, and reuse it on every call instead or creating a new one every time?

Reply:
Yes, a pointer to a SourceIPExtractor is passed in as a parameter now.